pkg/restore/restore.go

```diff
@@ -864,16 +865,20 @@ func (ctx *context) restoreItem(obj *unstructured.Unstructured, groupResource sc
 			}
 
 			if shouldRenamePV {
-				// give obj a new name, and record the mapping between the old and new names
-				oldName := obj.GetName()
-				newName, err := ctx.pvRenamer(oldName)
-				if err != nil {
-					addToResult(&errs, namespace, errors.Wrapf(err, "error renaming PV"))
-					return warnings, errs
+				if oldName == obj.GetName() {
+					// pvRestorer hasn't modified the PV name, we need to rename the PV
+					name, err = ctx.pvRenamer(oldName)
+					if err != nil {
+						addToResult(&errs, namespace, errors.Wrapf(err, "error renaming PV"))
+						return warnings, errs
+					}
+				} else {
+					// pvRestorer has renamed the PV, let's use the updated name
```

Comment:
please extend this comment to explain that a `VolumeSnapshotter` plugin could have modified the PV name via the `SetVolumeID` function (assuming this is accurate - this is how I'm reading it).

Reply:
Done.